### ISSUE TITLE
Server does not start

### ISSUE THREAD

**Issue by Brum8842** [AllTheMods/Server-Scripts#34]
Hello, I've been having trouble trying to get the server script to install forge and minecraft server files. I keep getting errors when it first runs.
I made a fresh install and here is the log https://pastebin.com/CuJ8SHZ8

this is what my screen looks like 
![image](https://user-images.githubusercontent.com/29263653/47247549-6374ec00-d3d2-11e8-90cd-c8d2ed19764c.png)

I've been trying to figure it out with no luck
thanks in advance

**Comment by squid-box** [comment]
I've had similar errors sure to badly formatted script and settings files - my suggestion is to double check settings.cfg to make sure it's properly formatted.

**Comment by Brum8842** [comment]
It is the default settings.cfg from the download, from what I'm seeing I don't see any problems.

**Comment by Glendening** [comment]
Navigate to the location of the serverstart.sh before running it in linux.

**Comment by Brum8842** [comment]
the serverstart.sh is in the folder with the settings.cfg

**Comment by Glendening** [comment]
Yes. What I am suggesting is to CD/ yourself into the location where serverstart is being launched from before launching it and see if the server loads properly.

**Comment by Brum8842** [comment]
That is the only place I've been launching it from, I don't launch it from outside the directory.

**Comment by e1nh4nd3r** [comment]
Can you provide a gist of the `serverstart.log` file so that we can see the environment variables?  There's an unexpected value or something that's missing from the server start script, which is why it's throwing all those errors about "integer value".

**Comment by Brum8842** [comment]
The pastebin I linked in my first post is the serverstart.log

**Comment by e1nh4nd3r** [comment]
Oh wow, I totally didn't see that.  Thanks, looking now.

**Comment by Glendening** [comment]
If you replace serverstart.sh and settings.cfg with fresh ones from the zip of the mod pack, and run serverstart.sh you are met with the same error?

**Comment by Brum8842** [comment]
yes what I posted was a very fresh unzip of the script files.

**Comment by Glendening** [comment]
Is this a local machine you have access to or a hosted server?

**Comment by Brum8842** [comment]
It is local, it is using Ubuntu 18.04.1

**Comment by Glendening** [comment]
Can you provide a pastebin of the settings.cfg file?

**Comment by Brum8842** [comment]
https://pastebin.com/kgj04C7i

It should be the same file from the zip folder, I did not change anything in it.

**Comment by FallenEagle1** [comment]
Given that this modpack seems to run an older version of forge, I am going to assume that there is an older version of the script that is being used as well. What I would suggest is replacing the serverstart.sh file with a newer version of it, that potentially would have fixes, this maybe being one of them. To make things easier, here is the newest version of the file. 
[ServerStart.zip](https://github.com/AllTheMods/Server-Scripts/files/2497774/ServerStart.zip)

**Comment by Glendening** [comment]
Brum, for research purposes, can you tell us what modpack you are running?

**Comment by Lartza** [comment]
The integer expression error comes from the fact that https://github.com/AllTheMods/Server-Scripts/blob/master/ServerStart.sh#L162 should be `if [ "${IGNORE_OFFLINE}" = "1;" ]; then`

Here's a serverstart.log too https://gist.github.com/Lartza/d545e2676e9cd8ae66409575f6fa007d

**Comment by Glendening** [comment]
For some reason it seems like the script is reading DISABLE as a url rather then as the default to be ignored. I wonder when this started and why more people aren't having issues?

**Comment by Lartza** [comment]
Probably because packs ship an installer.jar so the script rarely tries to download one.

**Comment by Glendening** [comment]
Just did a test
Replacing 
`if [ "${FORGEURL}" = "DISABLE" ]; then`
With 
`if [ "${FORGEURL}" = DISABLE ]; then`
Allows the script to connect, obtain files, and start the server.

**Comment by Yoosk** [comment]
Can you download [this](https://files.minecraftforge.net/maven/net/minecraftforge/forge/1.12.2-14.23.2.2625/forge-1.12.2-14.23.2.2625-installer.jar) forge installer, place it in same place as serverstart.sh, then run serverstart.sh?

**Comment by Lartza** [comment]
Mhh I just "fixed" it... https://gist.github.com/Lartza/f296fa4f14931a8165d7b1becccf214f#file-gistfile1-txt-L42

**Comment by Glendening** [comment]
So something else odd is going on here because if I take the serverstart.sh with its config, drop it in a folder and Bash StartServer.sh it works exactly as intended. So something else is going on here.

**Comment by Glendening** [comment]
Did some testing. Removing the 2 spaces (line 19 as well) from the .cfg allows the script to find the right URL and download the files but it looks like there are still issues.

**Comment by Lartza** [comment]
Because there are semicolons in that url

**Comment by Glendening** [comment]
Removed ; from line 16 and 17 of .cfg URL resolved and files downloaded successfully.

**Comment by Glendening** [comment]
Removed ALL semi-colons that are AFTER entries in the .cfg EXCEPT the very last one after DISABLE
Script loads, downloads files, starts server successfully.

**Comment by Brum8842** [comment]
Ok so I updated my settings.cfg to remove the semi-colons like Glendening mentioned and I also updated my ServerStart.sh file with adding the semi-colon after disable on line 76. I was able to get to the Starting install of Forge/Minecraft server binaries at the URL but that never downloaded, I noticed that it was just using Http and not Https so I updated line 77 to use Https and re-ran the script and was able to download the server binaries and start the server.
